airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -73,6 +74,31 @@ def update_state_for_stream(self, stream_name: str, namespace: Optional[str], va
         stream_descriptor = HashableStreamDescriptor(name=stream_name, namespace=namespace)
         self.per_stream_states[stream_descriptor] = AirbyteStateBlob.parse_obj(value)
 
+    def create_state_message(self, stream_name: str, namespace: Optional[str]) -> AirbyteMessage:
+        """
+        Generates an AirbyteMessage using the current per-stream state of a specified stream
+        :param stream_name: The name of the stream for the message that is being created
+        :param namespace: The namespace of the stream for the message that is being created
+        :return: The Airbyte state message to be emitted by the connector during a sync
+        """
+        hashable_descriptor = HashableStreamDescriptor(name=stream_name, namespace=namespace)
+        stream_state = self.per_stream_states.get(hashable_descriptor, AirbyteStateBlob())
```

Comment:
Is it problematic if `self.per_stream.states.get returns` None?

Reply:
Could we use `get_stream_state` method?

Reply:
i think we could actually simplify this and just do:
```
stream_state = self.per_stream_states.get(hashable_descriptor, AirbyteStateBlob()) or AirbyteStateBlob()
```

get will return none if it doesn't exist and if the value is none, we'll get the empty state blob anyway. That way we don't have to parse the dictionary back to another blob